src/test/java/GoServletTest.java

```diff
@@ -0,0 +1,254 @@
+  private LatLng streetAddressCoordinates = new LatLng(0, 0);
+  private GeocodingResult restaurantGeocodingResult = new GeocodingResult();
+
+  private Duration longerDuration = new Duration();
+  private DirectionsLeg longerLeg = new DirectionsLeg();
+  private DirectionsRoute longerRoute = new DirectionsRoute();
+  private DirectionsResult longerResult = new DirectionsResult();
+
+  private Duration shorterDuration = new Duration();
+  private DirectionsLeg shorterLeg = new DirectionsLeg();
+  private DirectionsRoute shorterRoute = new DirectionsRoute();
+  private DirectionsResult shorterResult = new DirectionsResult();
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    directionsClientFactory = Mockito.mock(DirectionsClientFactory.class);
+    directionsClient = Mockito.mock(DirectionsClient.class);
+    placesClientFactory = Mockito.mock(PlacesClientFactory.class);
+    placesClient = Mockito.mock(PlacesClient.class);
+    tasksClientFactory = Mockito.mock(TasksClientFactory.class);
+    tasksClient = Mockito.mock(TasksClient.class);
+    geocodingClientFactory = Mockito.mock(GeocodingClientFactory.class);
+    geocodingClient = Mockito.mock(GeocodingClient.class);
+
+    streetAddressGeocodingResult.geometry = streetAddressGeometry;
+    streetAddressGeocodingResult.geometry.location = streetAddressCoordinates;
+    streetAddressGeocodingResult.types = new AddressType[] {AddressType.STREET_ADDRESS};
+    restaurantGeocodingResult.types = new AddressType[] {AddressType.RESTAURANT};
+
+    longerDuration.inSeconds = 10;
+    longerLeg.duration = longerDuration;
+    longerRoute.legs = new DirectionsLeg[] {longerLeg};
+    longerResult.routes = new DirectionsRoute[] {longerRoute};
+
+    shorterDuration.inSeconds = 5;
+    shorterLeg.duration = shorterDuration;
+    shorterRoute.legs = new DirectionsLeg[] {shorterLeg};
+    shorterResult.routes = new DirectionsRoute[] {shorterRoute};
+
+    servlet =
+        new GoServlet(
+            directionsClientFactory,
+            placesClientFactory,
+            tasksClientFactory,
+            geocodingClientFactory,
+            API_KEY);
+
+    Mockito.when(directionsClientFactory.getDirectionsClient(API_KEY)).thenReturn(directionsClient);
+    Mockito.when(placesClientFactory.getPlacesClient(API_KEY)).thenReturn(placesClient);
+    Mockito.when(tasksClientFactory.getTasksClient(Mockito.any())).thenReturn(tasksClient);
+    Mockito.when(authenticationVerifier.verifyUserToken(Mockito.anyString()))
+        .thenReturn(AUTHENTICATION_VERIFIED);
+    Mockito.when(geocodingClientFactory.getGeocodingClient(API_KEY)).thenReturn(geocodingClient);
+  }
+
+  @Test
+  public void separateWaypoints() throws Exception {
+    // Street address should be converted into coordinates, restaurant should be converted into
+    // place type
+    List<String> waypoints = ImmutableList.of("street address", "restaurant");
```

Comment:
Make this a constant, since `streetAddressGeocodingResult` is already a constant. This list, as well as its contents, have semantic value

Reply:
Yep it's used twice too, thanks!

Reply:
`RESOLVED`